appserver/payara-appserver-modules/microprofile/opentracing/src/main/java/fish/payara/microprofile/opentracing/jaxrs/OpenTracingApplicationEventListener.java

```diff
@@ -88,9 +88,14 @@ public void postConstruct() {
             LOG.config("Default base service locator is null, JAX-RS server tracing is disabled.");
             return;
         }
+        final InvocationManager invocationManager = serviceLocator.getService(InvocationManager.class);
         this.requestTracing = serviceLocator.getService(RequestTracingService.class);
         this.openTracing = serviceLocator.getService(OpenTracingService.class);
-        this.applicationName = this.openTracing.getApplicationName(serviceLocator.getService(InvocationManager.class));
+        if (invocationManager == null || this.openTracing == null) {
+            this.applicationName = null;
+        } else {
+            this.applicationName = this.openTracing.getApplicationName(invocationManager);
```

Comment:
I don't know if this is a personal preference but I'd use a conditional if I had two cases to initialise the same variable. 
```java
this.applicationName = <test>
  ? <a>
  : <b>
```
Makes it more clear that there is a section of code and after the variable is initialised to a value.

Reply:
Hmm. It is very similar, but I looked at it after another week and the if/else was optically a bit more straightforward. I use conditionals most time only if it is single true/false condition. More complicated logic => method/function. Two lines ok, three lines warning sign :-)

```
 this.applicationName = invocationManager == null || this.openTracing == null ? null
            : this.openTracing.getApplicationName(invocationManager);
```